File: MakeBingoGameStream/BingoGame/src/BingoGame_MultiplePlayers/Lottery.java
package BingoGame_MultiplePlayers;

import java.util.ArrayList;
import java.util.Collections;
import java.util.List;

public class Lottery {
	private List<Integer> storedNumbers = new ArrayList<>();

	// コンストラクタ
	Lottery() {
		for (int i = 1; i <= 100; i++) {
			this.storedNumbers.add(i);
		}
	}

	// 当選番号を一個ずつ返すメソッド
	public int lottery() {
		Collections.shuffle(this.storedNumbers);
		int lotteryNumber = this.storedNumbers.get(0);
		this.storedNumbers.remove(0);
		return lotteryNumber;
	}
}

File: BingoGame_Ruby/BingoGame/lottery.rb
class Lottery
  def initialize
    @stored_numbers = []
    100.times do |i|
      @stored_numbers << (i + 1)
    end
  end

  # 当選番号を1個ずつ返すメソッド
  def lottery
    @stored_numbers.shuffle!
    lottery_number = @stored_numbers.at(0)
    @stored_numbers.shift
    lottery_number
  end

end